File: lib/src/Models/topic/topic.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:hive/hive.dart';

import 'package:learn_philosophy_app/src/Models/quiz/quiz.dart';
import '../site/site.dart';


part 'topic.freezed.dart';
part 'topic.g.dart';

@unfreezed 
class Topic with _$Topic{
  @HiveType(typeId: 4, adapterName: 'TopicAdapter')
  factory Topic({
    @HiveField(0) required int topicId,
    @HiveField(1) required String title,
    @HiveField(2) required String description,
    @HiveField(3) List<String>? categories,
    @HiveField(4) Quiz? quiz,
    @HiveField(5) @Default([]) List<Site> sites 
  }) = _Topic;

  factory Topic.fromJson(Map<String, Object?> json) => _$TopicFromJson(json);
}

File: api/PhilosophyLearnAppAPI/Models/Topic.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PhilosophyLearnAppAPI.Models
{

    public record Topic
    {
        [Key]
        public int TopicId { get; init; }

        [Required]
        [MaxLength(50)]
        public string Title { get; init; } = string.Empty;

        [Required]
        public string Description { get; init; } = string.Empty;

        public string[] Categories { get; init; } = [];

        public virtual ICollection<Site> Sites { get; init; } = [];
        public virtual Quiz Quiz { get; init; } = new Quiz();
    }
}
